common-libs/common/src/services/DidDocumentService/elem-lib/func.ts

```diff
@@ -38,12 +38,13 @@ export const getDidUniqueSuffix = (operation: any) => {
 }
 
 // NOTE check is signatures are the same as sidetree's (see NEP-334)
-export const signEncodedPayload = (encodedHeader: string, encodedPayload: string, privateKey: string) => {
+export const signEncodedPayload = (
+  encodedHeader: string,
+  encodedPayload: string,
+  signFn: (payload: Buffer) => Buffer,
+): Buffer => {
```

Comment:
Seems like it could be `getEncodedPayloadHash` instead, with one less level of indirection.

So that it would be called as
```
const hash = getEncodedPayloadHash(...)
const signed = sign(hash, ...)
```
instead of
```
const signed = signEncodedPayload(..., (hash) => sign(hash, ...))
```

Reply:
Moved the functions in op to DidElemDocument where they have access to the vault